frontend/src/components/datasets/DatasetTable.tsx

```diff
@@ -1,166 +1,70 @@
-    const page = datasetsState.page
+	const pagedItems = useAppSelector(selectDatasetsTable)
+	const { filters } = pagedItems
 
-    const dispatch = useAppDispatch()
-    const dispatchSetFilter = useCallback((...args) => dispatch(setFilter(...args)), [dispatch])
-    const dispatchSetFilterOption = useCallback((...args) => dispatch(setFilterOption(...args)), [dispatch])
-    const dispatchClearFilters = useCallback((...args) => dispatch(clearFilters(...args)), [dispatch])
-    const dispatchListFilter = useCallback((...args) => dispatch(listFilter(...args)), [dispatch])
-    const dispatchListTable = useCallback((...args) => dispatch(listTable(...args)), [dispatch])
-    const dispatchSetSortBy = useCallback((...args) => dispatch(setSortBy(...args)), [dispatch])
+	const callbacks = usePagedItemsActionsCallbacks(DatasetsTableActions)
+	useEffect(() => {
+		if (run_name) {
+			callbacks.setFilterCallback(DATASET_FILTER_KEYS.RUN, DATASET_FILTER_DEFINITIONS[DATASET_FILTER_KEYS.RUN])
+		}
+	}, [callbacks, run_name])
 
-    const columns = getTableColumns().map(c => Object.assign(c, getFilterProps(
-        c,
-        DATASET_FILTERS,
-        filters,
-        dispatchSetFilter,
-        dispatchSetFilterOption
-    )))
+	const tableColumns = useMemo(() => {
+		const columns = [
+			DATASET_COLUMN_DEFINITIONS.ID,
+			...(run_name ? [] : [DATASET_COLUMN_DEFINITIONS.RUN]),
+			DATASET_COLUMN_DEFINITIONS.PROJECT,
+			DATASET_COLUMN_DEFINITIONS.LANE,
+			DATASET_COLUMN_DEFINITIONS.VALIDATION_STATUS,
+			DATASET_COLUMN_DEFINITIONS.READSETS_RELEASED,
+			DATASET_COLUMN_DEFINITIONS.LATEST_RELEASE_UPDATE,
+			DATASET_COLUMN_DEFINITIONS.LATEST_VALIDATION_UPDATE
+		]
+		return columns
+	}, [run_name])
 
-    const nFilters = getNFilters(filters)
+  	const tweakedColumns = useFilteredColumns(
+          tableColumns,
+          DATASET_FILTER_DEFINITIONS,
+          DATASET_FILTER_KEYS,
+          filters,
+          callbacks.setFilterCallback,
+          callbacks.setFilterOptionsCallback
+	)
 
-    return <>
-        {
-            run_name
-                ? <FilteredList
-                    description={DATASET_FILTERS}
-                    columns={columns.filter((col) => col.dataIndex !== 'run_name')}
-                    listFilter={dispatchListFilter}
-                    items={filteredDatasets}
-                    itemsByID={datasetsById}
-                    totalCount={totalFilteredCount}
-                    filterID={run_name}
-                    filterKey={DATASET_FILTERS.run_name.key}
-                    isFetching={isFetching}
-                    page={page}
+	const getDataObjectsByID = useItemsByIDToDataObjects(selectDatasetsByID, dataset => {return {dataset}})
+		return(
+			<>
+                <FiltersBar filters={filters} clearFilters={callbacks.clearFiltersCallback}/>
+                <PagedItemsTable<ObjectWithDataset>
+                    columns={tweakedColumns}
                     expandable={ExpandableTableDatasetComments()}
```

Comment:
It's likely not because of your changes but could you tweak how the comment is displayed there is some issue there. There is too much space below the comments and not enough at the top. The empty icon (when there is no comment) get clipped at the top.